src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -180,11 +175,20 @@ protected AbstractEntityAIBasic(@NotNull final J job)
                 /**
                  * Called when the citizen saturation falls too low.
                  */
-                new AITarget(() -> (worker.getCitizenData().getSaturation() <= EntityCitizen.HIGH_SATURATION
-                        && !job.hasCheckedForFoodToday()) || worker.getCitizenData().getSaturation() <= 0, this::searchForFood)
+                new AITarget(this::shouldGetFood, this::searchForFood)
         );
     }
 
+    /**
+     * Check if the citizen should get food, meaning, check if he checked for food today already and check if his saturation is decent.
+     * @return true if he should go search for food.
+     */
+    private boolean shouldGetFood()
+    {
+        return (worker.getCitizenData().getSaturation() <= EntityCitizen.HIGH_SATURATION
```

Comment:
can you format this with line breaks so that each path is easily followable?
Each && and each || deserve a new line and () need intendation. Our formatter should handle that :)